diff --git a/CMakeLists.txt b/CMakeLists.txt
index 181387b0..356bf32c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,11 @@ include(CheckSymbolExists)
 include(CheckIncludeFile)
 include(CheckStructHasMember)

+set(CMAKE_MODULE_PATH
+  ${CMAKE_MODULE_PATH}
+  "${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules"
+)
+
 # osx specific
 set(CMAKE_MACOSX_RPATH TRUE)

@@ -81,7 +86,7 @@ endif()
 option(GEN_LANGUAGE_BINDINGS "Enable library bindings for different languages." OFF)

 option(GEN_CPP_BINDINGS "Enable C++ bindings." ON)
-option(BUILD_CPP_EXAMPLES "Enable C++ example application compilation." ON)
+option(BUILD_CPP_EXAMPLES "Enable C++ example application compilation." OFF)

 option(GEN_PYTHON_BINDINGS "Enable Python bindings." ON)
 option(ENABLE_PYTHON_TESTS "Enable Python tests." ON)
@@ -206,13 +211,13 @@ add_executable(sysrepo-plugind ${SYSREPOPLUGIND_SRC})
 target_link_libraries(sysrepo-plugind sysrepo dl)

 # installation
-install(TARGETS sysrepo DESTINATION ${CMAKE_INSTALL_LIBDIR})
-install(FILES ${PROJECT_SOURCE_DIR}/src/sysrepo.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(TARGETS sysrepo DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT sysrepo)
+install(FILES ${PROJECT_SOURCE_DIR}/src/sysrepo.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} COMPONENT sysrepo)
 install(FILES ${PROJECT_SOURCE_DIR}/src/utils/values.h ${PROJECT_SOURCE_DIR}/src/utils/xpath.h
-        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/sysrepo)
-install(TARGETS sysrepoctl sysrepocfg sysrepo-plugind DESTINATION ${CMAKE_INSTALL_BINDIR})
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/sysrepo COMPONENT sysrepo)
+install(TARGETS sysrepoctl sysrepocfg sysrepo-plugind DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT sysrepo)

-install(FILES "${PROJECT_BINARY_DIR}/libsysrepo.pc" DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+install(FILES "${PROJECT_BINARY_DIR}/libsysrepo.pc" DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig" COMPONENT sysrepo)

 # doxygen documentation
 find_package(Doxygen)
@@ -266,3 +271,9 @@ add_custom_target(sr_clean
 add_custom_target(uninstall "${CMAKE_COMMAND}" -P "${CMAKE_MODULE_PATH}/uninstall.cmake")
 add_custom_target(uninstall_with_repo "${CMAKE_COMMAND}" -P "${CMAKE_MODULE_PATH}/uninstall.cmake"
     COMMAND rm -rf ${REPOSITORY_LOC})
+
+ include(Packaging)
+ if(CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR)
+     include(Packager)
+     make_packages()
+ endif()
\ No newline at end of file
